include/circt/Dialect/SV/SVAttributes.td

```diff
@@ -53,42 +53,41 @@ def SVAttributeAttr : AttrDef<SVDialect, "SVAttribute"> {
     For more information, refer to Section 5.12 of the SystemVerilog (1800-2017)
     specification.
   }];
-  let parameters = (ins "::mlir::StringAttr":$name,
-                        OptionalParameter<"::mlir::StringAttr">:$expression);
+  let parameters = (ins
+    "::mlir::StringAttr":$name,
+    OptionalParameter<"::mlir::StringAttr">:$expression,
+    DefaultValuedParameter<
+      "mlir::BoolAttr", "mlir::BoolAttr::get($_ctxt, false)">:$emitAsComment
+  );
 
-  let assemblyFormat = [{ `<` $name ( `=` $expression^ )? `>` }];
+  let builders = [
+    AttrBuilder<(ins "::llvm::StringRef":$name,
+                     CArg<"bool", "false">: $emitAsComment), [{
+      return get(context, StringAttr::get(context, name), StringAttr(),
+                 BoolAttr::get(context, emitAsComment));
+    }]>,
+    AttrBuilder<(ins "::mlir::StringAttr":$name,
+                     CArg<"bool", "false">: $emitAsComment), [{
+      return get(context, name, StringAttr(),
+                 BoolAttr::get(context, emitAsComment));
+    }]>,
+    AttrBuilder<(ins "::llvm::StringRef":$name,
+                     "::llvm::StringRef":$expression,
+                     CArg<"bool", "false">: $emitAsComment), [{
+      return get(context, StringAttr::get(context, name),
+                 StringAttr::get(context, expression),
+                 BoolAttr::get(context, emitAsComment));
+    }]>,
+    AttrBuilder<(ins "::mlir::StringAttr":$name,
+                     "::mlir::StringAttr":$expression,
+                     CArg<"bool", "false">: $emitAsComment), [{
+      return get(context, name, expression,
+                 BoolAttr::get(context, emitAsComment));
+    }]>
+  ];
```

Comment:
It's unfortunate that MLIR cannot generate these ;)

Reply:
Yeah. Just listing the entire combinatorial explosion of param combinations isn't really a good approach. In Rust land there's the "builder pattern" that people use, where you create a helper struct for collecting fields in multiple steps:
```c++
struct Builder {
  Builder &name(StringRef arg) { return name(arg); }
  Builder &name(StringAttr arg) { nameArg = StringAttr::get(arg); return *this; }
  Builder &expression(StringRef arg) { return expression(arg); }
  Builder &expression(StringAttr arg) { expressionArg = StringAttr::get(arg); return *this; }
  Builder &emitAsComment(bool arg = true) { return emitAsComment(arg); }
  Builder &emitAsComment(BoolAttr arg) { emitAsCommentArg = BoolAttr::get(arg); return *this; }
  SVAttributeAttr build() {
    return SVAttributeAttr::get(nameArg, expressionArg, emitAsCommentArg);
  }

private:
  StringAttr nameArg;
  StringAttr expressionArg;
  BoolAttr emitAsCommentArg;
};

// Usage:
Builder().name("bla").build()
Builder().name(strAttr).expression("foobar").emitAsComment(boolAttr).build()
```